src/main/java/controllers/CommandLineInterface.java

```diff
@@ -173,20 +205,79 @@ public static Schedule promptImportSchedule() {
         return userFilters;
     }
 
+    /**
+     * Confirms whether user wants to generate all schedules or use one by one generation.
+     */
+    public void confirmOneByOneGeneration() {
+        Scanner scanner = new Scanner(System.in);
+        System.out.println(
+                "Would you like to generate schedules one by one?\n"
+                + "Your schedule will be populated with only one course at a time to allow for specific time slot"
+                + " selection.\n"
+                + "1/0 for Y/N. \n"
+                + "Non-integer inputs will quit selection."
+        );
+        while (scanner.hasNextInt()) {
+            int input = scanner.nextInt();
+            if (input == 1 || input == 0) {
+                this.generationMode = input;
+                return;
+            }
+        }
+    }
+
+    /** Asks user to select the next base schedule for permutation in Scheduler.
+     *
+     * @param userSchedules the schedules meeting previous user specifications with one more course being permuted
+     * @return if the user selects a schedule around
+     */
+    public Schedule promptUserBaseSchedule(List<Schedule> userSchedules) {
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Please select the schedule around which you want other time slots to be populated."
+        );
+        Schedule nextSchedule = this.displayUserSchedules(userSchedules);
+
+        if (nextSchedule == null) {
+            System.out.println("You have not selected a schedule.\n" +
+                    "The scheduler will generate all available schedules meeting previous specifications."
+            );
+        }
+        System.out.println(
+                "Would you like to continue one-by-one generation?\n"
+                        + "1/0 for Y/N. \n"
+                        + "Non-integer inputs will quit selection."
+        );
+        while (scanner.hasNextInt()) {
+            int input = scanner.nextInt();
+            if (input == 1 || input == 0) {
+                this.generationMode = input;
+                return nextSchedule;
+            }
+        }
+        return nextSchedule;
+    }
+
+
+
     /**
      * Outputs schedules meeting user criteria. User can navigate through schedules and save them.
      *
      * @param userSchedules schedules meeting filter criteria
+     *
+     * attribute 'generationMode' is used in this method with
+     * 0 -> returning a Schedule is not an option
+     * 1 -> returning a Schedule is an option
+     * Note: returning a schedule is required in 1 by 1 generation
      */
-    public static void displayUserSchedules(List<Schedule> userSchedules) {
+    public Schedule displayUserSchedules(List<Schedule> userSchedules) {
```

Comment:
I get that you need to return `Schedule`, but it kinda doesn't make sense for a method called `displayUserSchedules`. However, I honestly can't figure out a better name tho.